Sources/protoc-gen-swift/GeneratorOptions.swift

```diff
@@ -130,5 +130,24 @@ class GeneratorOptions {
     }
 
     self.implementationOnlyImports = implementationOnlyImports
+
+    // ------------------------------------------------------------------------
+    // Now do "cross option" validations.
+
+    if self.implementationOnlyImports && self.visibility != .internal {
+      throw GenerationError.message(message: """
+        Cannot use @_implementationOnly imports when the proto visibility is public or package.
+        Either change the visibility to internal, or disable @_implementationOnly imports.
+        """)
+    }
+
+    // The majority case is that if `self.protoToModuleMappings.hasMappings` is
+    // true, then `self.visibility` should be either `.public` or `.package`.
+    // However, it is possible for someone to put the "leaf" proto files in a
```

Comment:
The word "leaf" here is confusing to me; I would think of the leaf files as the ones without any dependencies and would think of your "leaf" as the "root".

It might be clearer with just a concrete example ("A" depends on "B", B is public but A is internal).

Reply:
Revised the text including an example.